lib/node_modules/@stdlib/blas/base/cscal/lib/cscal.js

```diff
@@ -0,0 +1,96 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex64' );
+
+
+// MAIN //
+
+/**
+* Scales a complex single-precision floating-point vector by a single-precision floating point constant.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Complex64Array} cx - constant
+* @param {Complex64Array} x - input array
+* @param {integer} strideX - `x` stride length
+* @returns {Complex64Array} output array
+*
+* @example
+* var Complex64Array = require( '@stdlib/array/complex64' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* var x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+* var cx = new Complex64Array( [ 2.0, 2.0 ] );
+*
+* cscal( x.length, cx, x, 1 );
+*
+* var z = y.get( 0 );
+* // returns <Complex64>
+*
+* var re = realf( z );
+* // returns -2.0
+*
+* var im = imagf( z );
+* // returns 6.0
+*/
+function cscal( N, cx, x, strideX ) {
+	var viewCX;
+	var viewX;
+	var tmpr;
+	var tmpi;
+	var sx;
+	var ix;
+	var i;
+
+	if ( N <= 0 || strideX <= 0 ) {
+		return x;
+	}
+	viewX = reinterpret( x, 0 );
+	viewCX = reinterpret(cx, 0);
+	if ( strideX === 1 ) {
+		// code for stride equal to `1`...
+		for ( i = 0; i < N*2; i += 2 ) {
+			tmpr = viewX[ i ];
+			tmpi = viewX[ i+1 ];
+			viewX[ i ] = viewCX[ 0 ] * tmpr - viewCX[ 1 ] * tmpi;
```

Comment:
For multiplication of two complex numbers in JS, use https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/ops/cmulf.